localstack/services/apigateway/invocations.py

```diff
@@ -60,45 +83,50 @@ def __init__(self, context: ApiInvocationContext, store: ApiGatewayStore = None)
             raise NotFound()
         self.rest_api_container = container
 
-    def is_request_valid(self) -> bool:
+    def validate_request(self) -> None:
+        """
+        :raises BadRequestParameters if the request has required parameters which are not present
+        :raises BadRequestBody if the request has required body validation with a model and it does not respect it
+        :return: None
+        """
         # make all the positive checks first
         if self.context.resource is None or "resourceMethods" not in self.context.resource:
-            return True
+            return
 
         resource_methods = self.context.resource["resourceMethods"]
         if self.context.method not in resource_methods and "ANY" not in resource_methods:
-            return True
+            return
 
         # check if there is validator for the resource
         resource = resource_methods.get(self.context.method, resource_methods.get("ANY", {}))
         if not (resource.get("requestValidatorId") or "").strip():
-            return True
+            return
 
         # check if there is a validator for this request
         validator = self.rest_api_container.validators.get(resource["requestValidatorId"])
         if not validator:
-            return True
+            return
 
-        # are we validating the body?
-        if self.should_validate_body(validator):
-            is_body_valid = self.validate_body(resource)
-            if not is_body_valid:
-                return is_body_valid
+        if self.should_validate_request(validator) and (
+            param_errors := self.validate_parameters_and_headers(resource)
+        ):
+            message = f"Missing required request parameters: [{', '.join(param_errors)}]"
+            raise BadRequestParameters(message=message)
 
-        if self.should_validate_request(validator):
-            is_valid_parameters = self.validate_parameters_and_headers(resource)
-            if not is_valid_parameters:
-                return is_valid_parameters
+        if self.should_validate_body(validator) and not self.validate_body(resource):
+            raise BadRequestBody(message="Invalid request body")
 
-        return True
+        return
 
-    def validate_body(self, resource):
+    def validate_body(self, resource) -> bool:
```

Comment:
should follow the same command name pattern, either use `is_body_valid` and returns a bool or `validate_body` and does return anything.

Reply:
yes you're right! I've renamed into `_is_body_valid`, but for the parameters and headers, as the logic needed is different, I'll rename it into `_get_missing_required_parameters`. Not fan of the naming but didn't get more inspiration.